src/spi.rs

```diff
@@ -482,8 +514,33 @@ where
         }
 
         // Do one last status register check before continuing
-        self.check_send().ok();
-        Ok(())
+        self.check_errors()
+    }
+}
+
+impl<SPI, SCKPIN, MISOPIN, MOSIPIN> ::embedded_hal::spi::FullDuplex<u16>
+    for Spi<SPI, SCKPIN, MISOPIN, MOSIPIN, SixteenBit>
+where
+    SPI: Deref<Target = SpiRegisterBlock>,
+{
+    type Error = Error;
+
+    fn read(&mut self) -> nb::Result<u16, Error> {
+        self.check_read()?;
+        Ok(self.read_u16())
+    }
+
+    fn send(&mut self, byte: u16) -> nb::Result<(), Error> {
+        // We want to transfer bidirectionally, make sure we're in the correct mode
+        self.set_bidi();
+
+        self.check_send()?;
+        self.send_u16(byte);
+
+        match self.check_errors() {
+            Ok(_) => Ok(()),
+            Err(e) => Err(nb::Error::Other(e)),
+        }
```

Comment:
Couldn't this be written more obviously as:

```suggestion
       self.check_errors().map_err(|e| Err(nb::Error::Other(e)))
```

Reply:
I thought of that, but it seems you can only use it to change the error, not the `Result` type wrapping everything (`Result` vs `nb::Result` here). I get this:
```
rror[E0308]: mismatched types
   --> src/spi.rs:482:9
    |
475 |     fn send(&mut self, byte: u8) -> nb::Result<(), Error> {
    |                                     --------------------- expected `core::result::Result<(), nb::Error<spi::Error>>` because of return type
...
482 |         self.check_errors().map_err(|e| Err(nb::Error::Other(e)))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `nb::Error`, found enum `core::result::Result`
```
But maybe I'm not doing something correctly ?

Reply:
Sorry, my bad. Cut'n'pasto. Of course it needs to be (corrected above, too):
```
self.check_errors().map_err(|e| nb::Error::Other(e))
```
`map_err()` automatically wraps the value returned from the closure in an `Err`, cf. here: https://doc.rust-lang.org/src/core/result.rs.html#592

Reply:
I believe that the `map_err` isn't needed at all. Check for example the `check_send` method, where `check_errors` is called directly with the `?` operator. There might be an automatic Into trait implementation for it.

Reply:
It works with `?` yes (and I was surprised), but not if using the statement for returning (ie. without `?`). I'll fix the `map_err` except if you have something better in mind ?

Reply:
> I believe that the `map_err` isn't needed at all. Check for example the `check_send` method, where `check_errors` is called directly with the `?` operator. There might be an automatic Into trait implementation for it.

Depends on the signature. If we're replying the right `Result` type then we can just use it directly. Indeed we can also use `?` if we have proper conversion implementations available, however IIRC using `?` as the last operation in a function is frowned upon and linted by clippy. ;)

Reply:
Oh, sorry, I missed that it was the last command in function. LGTM then with `map_err`.